custom_components/ultimaker/camera.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import subprocess
+import re
+import aiohttp
+from homeassistant.components.camera import Camera, CameraEntityFeature
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN, COORDINATOR
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_mac_from_ip(ip):
```

Comment:
Don't assume that every system on hich home assistant runs has these os utils installed.
It might be better to use an existing and dedicated Python library for this. A quick internet search tells me tat getmac https://pypi.org/project/getmac/ is an interresting candidate.